ios/RNSScreen.mm

```diff
@@ -964,6 +989,73 @@ - (void)viewDidLayoutSubviews
   }
 }
 
+- (void)dismissViewControllerAnimated:(BOOL)flag completion:(void (^)())completion
+{
+  [super dismissViewControllerAnimated:flag completion:completion];
+  BOOL isPresentedAsNativeModal = self.screenView.isPresentedAsNativeModal;
+
+  [self recalculateHeaderHeightIsModal:isPresentedAsNativeModal];
+}
+
+- (CGFloat)getCalculatedHeaderHeight:(BOOL)isModal
+{
+  CGFloat navbarHeight = self.navigationController.navigationBar.frame.size.height;
+
+  // In case where screen is a modal, we want to calculate it's childViewController's height
+  if (isModal && self.childViewControllers.count > 0 && self.childViewControllers[0] != nil) {
+    UINavigationController *childNavCtr = self.childViewControllers[0];
+    navbarHeight = childNavCtr.navigationBar.frame.size.height;
+  }
+
+  return navbarHeight;
+}
+
+- (CGSize)getCalculatedStatusBarHeightIsModal:(BOOL)isModal
+{
+#if !TARGET_OS_TV
+  BOOL isGrabbableModal = isModal && ![RNSScreenModal isFullscreenModal:self.modalPresentationStyle];
+
+  // When modal is floating (we can grab its header), we don't want to calculate status bar in it.
+  // Thus, we return '0' as a height of status bar.
+  if (self.childViewControllers.count > 0 && self.childViewControllers[0] != nil && isGrabbableModal) {
+    return CGSizeMake(0, 0);
+  }
+
+  CGSize fallbackStatusBarSize = [[UIApplication sharedApplication] statusBarFrame].size;
+
+  if (@available(iOS 13.0, *)) {
+    CGSize primaryStatusBarSize = self.view.window.windowScene.statusBarManager.statusBarFrame.size;
+    if (primaryStatusBarSize.height == 0 || primaryStatusBarSize.width == 0)
+      return fallbackStatusBarSize;
+
+    return primaryStatusBarSize;
+  } else {
+    return fallbackStatusBarSize;
+  }
+#else
+  // On TVOS, status bar doesn't exist
+  return CGSizeMake(0, 0);
+#endif // !TARGET_OS_TV
+}
+
+- (void)recalculateHeaderHeightIsModal:(BOOL)isModal
+{
+  if ([RNSScreenModal isTransparentModal:self.modalPresentationStyle]) {
+    [self.screenView notifyHeaderHeightChange:0];
```

Comment:
I believe transparent modals both on Android and iOS have headers and yielding `0` isn't correct

Reply:
Modals on `iOS` do not have headers, but we discussed it with @tboba already. On `Android`, modal is exactly the same fragment as the normal screen, but with the transparent background iirc. Or am I wrong @kkafar ?